src/rtmp/rtmpclientconnection.cpp

```
@@ -82,8 +82,15 @@ int RTMPClientConnection::Connect(const char* server, int port, const char* app,
 
 	//If not found
 	if (!host)
+	{
 		//Error
+		if (listener)
+		{
+			listener->onError(this, RTMPClientConnection::ErrorCode::FailedToResolveURL);
```

Given that the Connect is a pure synchronous method, I would prefer to return the error code instead of firing an error event in this case.

---

Yes. I thought of that. In that case, we would need to change the return value to be something like 0 means success. Will look into it.

---

yes, maybe returning an enum could be a good idea. @mildsunrise wdyt?